sycl/source/detail/scheduler/commands.cpp

```diff
@@ -76,6 +76,33 @@ void emitInstrumentationGeneral(uint32_t StreamID, uint64_t InstanceID,
 }
 #endif
 
+RTDeviceBinaryImage *
+retrieveAMDGCNOrNVPTXKernelBinary(const DeviceImplPtr DeviceImpl,
+                                  const std::string &KernelName) {
+  const bool IsNvidia = DeviceImpl->getBackend() == backend::ext_oneapi_cuda;
+  const bool IsHIP = DeviceImpl->getBackend() == backend::ext_oneapi_hip;
+  if (!(IsNvidia || IsHIP))
+    throw sycl::exception(sycl::make_error_code(sycl::errc::runtime),
+                          "Unsupported backend" +
+                              codeToString(PI_ERROR_INVALID_VALUE));
```

Comment:
```suggestion
                          "Unsupported backend " +
                              codeToString(PI_ERROR_INVALID_VALUE));
```

Reply:
Should we `assert` here instead as this should never happen (we're programming it, right?)

Reply:
I have reverted to using (and adapting a bit) [already existing function](https://github.com/jchlanda/llvm/blob/jakub/jit_spec_const/sycl/include/sycl/detail/helpers.hpp#L261), so this is no longer applicable.